doc/design/topology.md

```diff
@@ -0,0 +1,119 @@
+# Topology Overview
```

Comment:
> Topology Overview

==>

> Design Doc: Add New Layers

to make the title of this document consistent with the title of this PR.